bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1547,12 +1552,36 @@ private void updateCSRLabel(JdkVersion version, Map<String, IssueTrackerIssue> c
                 existingApprovedCSR = true;
             }
         }
-        if (notExistingUnresolvedCSR && (!isCSRNeeded(comments) || existingApprovedCSR) && pr.labelNames().contains(CSR_LABEL)) {
+        if (notExistingUnresolvedCSR && (!isCSRNeeded(comments) || existingApprovedCSR) && newLabels.contains(CSR_LABEL)) {
             log.info("All CSR issues closed and approved for " + describe(pr) + ", removing CSR label");
             newLabels.remove(CSR_LABEL);
         }
     }
 
+    private void updateBackportCSRLabel(Map<Issue, Optional<IssueTrackerIssue>> regularIssuesMap, Map<String, IssueTrackerIssue> issueToCsrMap) {
+        if (newLabels.contains("backport") && !newLabels.contains("csr") && issueToCsrMap.isEmpty() && !isCSRManuallyUnneeded(comments)) {
+            boolean hasResolvedCSR = regularIssuesMap.values().stream()
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .map(Backports::csrLink)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .map(Link::issue)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .anyMatch(csrIssue -> csrIssue.state() == org.openjdk.skara.issuetracker.Issue.State.CLOSED &&
+                            csrIssue.resolution().map(res -> res.equals("Approved")).orElse(false));
+
+            if (hasResolvedCSR) {
+                newLabels.add("csr");
+                pr.addComment("At least one of the associated issues of this backport has a resolved CSR.\n" +
+                        "This backport might also need a CSR. `csr` label will be added to this PR.\n" +
+                        "Please go through the [CSR](" + CSR_PROCESS_LINK +
+                        ") process or using command `csr unneeded` to remove the CSR requirement.");
```

Comment:
There is no need to use single newlines, they will just be ignored when the comment is formatted anyway.

```suggestion
                pr.addComment("At least one of the issues associated with this backport has a resolved " +
                        "[CSR](" + CSR_PROCESS_LINK + ") for a different version. As this means that this " +
                        "backport may also need a CSR, the `csr` label is being added to this pull request " +
                        "to signal this potential requirement. The command `/csr unneeded` can be used to " + 
                        "remove the label in case a CSR is not needed.");
```

Reply:
If something goes wrong in CheckRun after posting this comment and before the labels are updated, we run the risk of posting this comment multiple times. We should add a marker in the comment and check that we haven't posted one before.

Reply:
Right, thanks for catching it!